plugins/modules/ipa_sudorule.py

```diff
@@ -384,12 +384,10 @@ def ensure(module, client):
                 client.sudorule_add_allow_command_group(name=name, item=cmdgroup)
 
         if deny_cmd is not None:
-            changed = category_changed(module, client, 'cmdcategory', ipa_sudorule) or changed
```

Comment:
`changed` still needs to be updated in this case, otherwise idempotency won't work correctly. (Same below.)

Reply:
apologizing for the incredibly late reply here.
Can you please explain what isn't working correctly? We have not noticed any idemptotency issues during our testing.

Reply:
When you modify the value of `deny_cmdgroup` or `deny_cmd`, `changed` should be `True` (both in check mode and regular execution). In the PR you're removing the code which updates `changed` in these cases, that's why I'm asking - maybe it's set already somewhere else? But maybe also not?

Reply:
Got it,  I'll need to look at how to ensure changed gets set safely. the problem we had is that category_changed on line 317 resets category_name to "none" anytime it is equal to "all" which may be desirable for cmd or cmd_group, but not for deny_cmd or deny_cmdgroup